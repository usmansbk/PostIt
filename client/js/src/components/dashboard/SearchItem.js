import React from 'react';
import Icon from '../common/Icon';
import '../../../../stylesheets/sass/components/SearchItem.scss'

export default ({userAvatar, username, isMember}) => {
	const style = {
		height: '24px',
		width: '24px',
	};

	return (
		<div className='row card-panel' id='search-item' >
			<div className='col s2'>
				<img src={userAvatar} alt={username} className='circle' style={style} />
			</div>
			<div className='col s10'>
				<div className='row'>
					<div className='col s6 valign-wrapper'>
						<span className='grey-text text-darken-2'>{username}</span>
					</div>
					<div className='col s6'>
						<Icon className='blue-text right'>person_add</Icon>
					</div>
				</div>
			</div>
		</div>
	);
}