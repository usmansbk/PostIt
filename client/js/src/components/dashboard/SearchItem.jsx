import React from 'react';
import Icon from '../common/Icon';
import '../../../../stylesheets/sass/components/SearchItem.scss'

export default ({ username, isMember, page, canAdmin, gid, onClick}) => {
  return (
    <div className='row card-panel' id='search-item' >
      <div className='col s12'>
        <div className='row'>
          <div className='col s6 valign-wrapper'>
            <Icon className='grey-text text-darken-2'>account_box</Icon>
            <span className='grey-text text-darken-2'>{username}</span>
          </div>
          {
            (page === 'Group' && canAdmin && !isMember) &&
            <div className='col s6'>
              <span title='Add user to this group'>
              <Icon className='green-text right' gid={gid} username={username} onClick={onClick}>add</Icon>
              </span>
            </div>
          }
        </div>
      </div>
    </div>
  );
}