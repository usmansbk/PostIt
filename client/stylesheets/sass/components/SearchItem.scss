#search-item {
	width: 100%;
	padding: 0 !important;
	margin: 0;
	height: 50px;
	div {
		height: 50px;
	};
	i {
		cursor: pointer;
	};
	&:hover {
		background-color:#cfd8dc;
	}
}