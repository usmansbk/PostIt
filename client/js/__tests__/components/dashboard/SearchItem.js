import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import SearchItem from '../../../src/components/dashboard/SearchItem';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const props = {
    page: null,
    canAdmin: false,
    isMember: false,
    add: false,
    addFailed: false,
    gid: 1,
    onClick: jest.fn()
  }

  const mockEvent = {};
  mockEvent.preventDefault = jest.fn();

  const wrapper = mount(<SearchItem {...props } />)

  return {
    props,
    wrapper,
    mockEvent
  }
}

describe('SearchItem', () => {
  it('should render component', () => {
    const component = renderer.create(<SearchItem />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })


  it('should not render add component', () => {
    const { wrapper, props } = setup();
    wrapper.setProps({
      page: 'Group',
      canAdmin: false,
      isMember: false
    });
    const addIcon = wrapper.find(`i[gid=${props.gid}]`)
    expect(addIcon.length).toBeFalsy();
  }) 

  it('should not render add component', () => {
    const { wrapper, props } = setup();
    wrapper.setProps({
      page: 'Group',
      canAdmin: true,
      isMember: true
    });
    const addIcon = wrapper.find(`i[gid=${props.gid}]`)
    expect(addIcon.length).toBeFalsy();
  })

  it('should render add component', () => {
    const { wrapper, props } = setup();
    wrapper.setProps({
      page: 'Group',
      canAdmin: true,
      isMember: false
    });
    const addIcon = wrapper.find(`i[gid=${props.gid}]`)
    expect(addIcon.length).toBeTruthy();
  })

  it('should call onClick', () => {
    const { wrapper, mockEvent, props } = setup();
    wrapper.setProps({
      page: 'Group',
      canAdmin: true,
      isMember: false
    });
    const addComponent = wrapper.find(`i[gid=${props.gid}]`)
    addComponent.simulate('click', mockEvent);
    expect(props.onClick).toBeCalled();
  })
})