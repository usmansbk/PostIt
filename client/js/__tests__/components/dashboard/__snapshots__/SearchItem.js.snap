// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchItem should render component 1`] = `
<div
  className="row card-panel"
  id="search-item"
>
  <div
    className="col s12"
  >
    <div
      className="row"
    >
      <div
        className="col s6 valign-wrapper"
      >
        <i
          className="material-icons grey-text text-darken-2"
        >
          account_box
        </i>
        <span
          className="grey-text text-darken-2"
        />
      </div>
    </div>
  </div>
</div>
`;
